### PR TITLE
Fix problem where the (Hadoop) filesystem was being found from the

### LINKED ISSUE #1324: blowup on direct runner with hdfs
` ./gatk-launch  CountReadsSpark --input hdfs:///user/akiezun/data/HCC1143_BL.bam -- --sparkRunner DIRECT`

gives me 
```
java.lang.IllegalArgumentException: Wrong FS: hdfs:/user/akiezun/data/HCC1143_BL.bam, expected: file:///
	at org.apache.hadoop.fs.FileSystem.checkPath(FileSystem.java:647)
	at org.apache.hadoop.fs.FileSystem.makeQualified(FileSystem.java:466)
	at org.apache.hadoop.fs.FilterFileSystem.makeQualified(FilterFileSystem.java:119)
	at org.broadinstitute.hellbender.engine.spark.datasources.ReadsSparkSource.getHeader(ReadsSparkSource.java:181)
	at org.broadinstitute.hellbender.engine.spark.GATKSparkTool.initializeReads(GATKSparkTool.java:284)
	at org.broadinstitute.hellbender.engine.spark.GATKSparkTool.initializeToolInputs(GATKSparkTool.java:264)
	at org.broadinstitute.hellbender.engine.spark.GATKSparkTool.runPipeline(GATKSparkTool.java:255)
	at org.broadinstitute.hellbender.engine.spark.SparkCommandLineProgram.doWork(SparkCommandLineProgram.java:36)
	at org.broadinstitute.hellbender.cmdline.CommandLineProgram.runTool(CommandLineProgram.java:98)
	at org.broadinstitute.hellbender.cmdline.CommandLineProgram.instanceMainPostParseArgs(CommandLineProgram.java:146)
	at org.broadinstitute.hellbender.cmdline.CommandLineProgram.instanceMain(CommandLineProgram.java:165)
	at org.broadinstitute.hellbender.Main.instanceMain(Main.java:66)
	at org.broadinstitute.hellbender.Main.main(Main.java:81)
```

### PR DESCRIPTION
**Pull request by tomwhite**
default configuration rather than the path.

Fixes #1324.

### CODE REVIEW

**Comment by lbergelson** [comment]
@tomwhite I added a unit test that checks if we can read from both `file://` and `hdfs://`.  It failed before your change but passes now.

**Comment by lbergelson** [comment]
back to you tom, could you make sure my test is sane?

File: src/test/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSourceUnitTest.java [hunk 1]
```diff
@@ -98,6 +102,34 @@ public void testPartitionSizing(){
...
+        final File bam = new File( getToolTestDataDir(), "reads.bam");
+        final File bai = new File( getToolTestDataDir(), "reads.bai");
+        MiniDFSCluster cluster = null;
+        try {
+            cluster = new MiniDFSCluster.Builder(new Configuration()).build();
+            final Path workingDirectory = cluster.getFileSystem().getWorkingDirectory();
+            final Path bamPath = new Path(workingDirectory,"flag_stat_hdfs.bam");
+            final Path baiPath = new Path(workingDirectory, "flag_stat_hdfs.bai");
+            cluster.getFileSystem().copyFromLocalFile(new Path(bam.toURI()), bamPath);
+            cluster.getFileSystem().copyFromLocalFile(new Path(bai.toURI()), baiPath);
```
Comment (droazen) [inline src/test/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSourceUnitTest.java]: Where do these files end up? Are they deleted on exit?

Comment (lbergelson) [inline src/test/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSourceUnitTest.java]: that's a good question... I was assuming yes... but on further inspection I don't know

File: src/test/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSourceUnitTest.java [hunk 2]
```diff
@@ -98,6 +102,34 @@ public void testPartitionSizing(){
         Assert.assertEquals(smallPartitions.partitions().size(), 2);
     }
 
+    @Test
+    public void testReadFromFileAndHDFS() throws IOException {
+        final File bam = new File( getToolTestDataDir(), "reads.bam");
```
Comment (droazen) [inline src/test/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSourceUnitTest.java]: Might want a less-generic name for this test input

**Comment by tomwhite** [comment]
Thanks for writing a test Louis! Looks fine to me.

File: src/test/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSourceUnitTest.java (hunk 1 above)
Comment (tomwhite) [inline src/test/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSourceUnitTest.java]: The entire mini cluster working directory is deleted on exit (see https://github.com/apache/hadoop/blob/trunk/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/MiniDFSCluster.java#L1918, note that the `base_dir` is in _build/test/data/dfs_ by default).

File: src/test/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSourceUnitTest.java (hunk 2 above)
Comment (tomwhite) [inline src/test/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSourceUnitTest.java]: Any suggestions?

Comment (droazen) [inline src/test/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSourceUnitTest.java]: Any name that makes it clear what the test data is being used for -- eg., `hdfs_test.bam`

**Comment by lbergelson** [comment]
I've updated the file names.  Going to merge when they pass.

**Pull request merged**